test/Mono.Linker.Tests.Cases/Inheritance.Interfaces/InterfaceVariants.cs

```diff
@@ -20,29 +20,63 @@ public static void Main ()
 			t = typeof (UninstantiatedPublicClassWithImplicitlyImplementedInterface);
 			t = typeof (UninstantiatedPublicClassWithPrivateInterface);
 
-			UninstantiatedPublicClassWithInterface.InternalStaticInterfaceMethodUsed ();
-			InstantiatedClassWithInterfaces.InternalStaticInterfaceMethodUsed ();
-
+			UninstantiatedPublicClassWithInterface.InternalStaticInterfaceMethodUsedThroughImplementation ();
+			InstantiatedClassWithInterfaces.InternalStaticInterfaceMethodUsedThroughImplementation ();
+			GenericMethodThatCallsInternalStaticInterfaceMethod
+				<ImplementsInternalStaticInterfaceWithInterfaceDefinitionUsedThroughGeneric> ();
+			typeof (InterfaceVariants).GetMethod ("GenericMethodThatCallsInternalStaticInterfaceMethod")
+				.MakeGenericMethod (typeof (ImplementsInternalStaticInterfaceWithInterfaceDefinitionNotUsedThroughGeneric))
+				.Invoke (null, null);
 			// Use all public interfaces - they're marked as public only to denote them as "used"
 			typeof (IPublicInterface).RequiresPublicMethods ();
 			typeof (IPublicStaticInterface).RequiresPublicMethods ();
 
 			var a = new InstantiatedClassWithInterfaces ();
 		}
 
+		[Kept]
+		internal static void GenericMethodThatCallsInternalStaticInterfaceMethod<T> () where T : IInternalStaticInterfaceWithInterfaceDefinitionUsed
+		{
+			T.InternalStaticInterfaceUsedThroughInterface ();
+		}
+
+		[Kept]
+		[KeptInterface (typeof (IInternalStaticInterfaceWithInterfaceDefinitionUsed))]
+		internal class ImplementsInternalStaticInterfaceWithInterfaceDefinitionUsedThroughGeneric : IInternalStaticInterfaceWithInterfaceDefinitionUsed
+		{
+			[Kept]
+			[KeptOverride (typeof (IInternalStaticInterfaceWithInterfaceDefinitionUsed))]
+			public static void InternalStaticInterfaceUsedThroughInterface ()
+			{
+			}
+			public static void UnusedMethod () { }
+		}
+
+		[Kept]
+		[KeptInterface (typeof (IInternalStaticInterfaceWithInterfaceDefinitionUsed))]
+		internal class ImplementsInternalStaticInterfaceWithInterfaceDefinitionNotUsedThroughGeneric : IInternalStaticInterfaceWithInterfaceDefinitionUsed
```

Comment:
I can't quite tell if this is meant to mean that the type is never used through a generic substitution. If so, that would mean that the `InternalStaticInterfaceUsedThroughInterface` method below has its override kept regardless.

I don't necessarily think this is the wrong behavior, but it's good to note. We might want to file a bug about a potential improvement in the future, if that's something we're interested in.